src/block/accordion/index.php

```diff
@@ -66,3 +66,74 @@ function stackable_load_accordion_frontend_polyfill_script() {
 	}
 	add_action( 'stackable/accordion/enqueue_scripts', 'stackable_load_accordion_frontend_polyfill_script' );
 }
+
+if ( ! class_exists( 'Stackable_Accordion_FAQ_Schema') ) {
+	class Stackable_Accordion_FAQ_Schema {
+		public $faq_entities = [];
+
+		function __construct() {
+			add_filter( 'render_block_stackable/accordion', array( $this, 'render_block_accordion_faq_schema' ), 10, 2 );
+			add_filter( 'wp_footer', array( $this, 'print_faq_schema' ) );
+		}
+
+		public function print_faq_schema() {
+			if ( count( $this->faq_entities ) ) {
+				echo
+				'<script type="application/ld+json">
+				{
+					"@context": "https://schema.org",
+					"@type": "FAQPage",
+					"mainEntity": [' . implode( ', ', $this->faq_entities ) . ']
+				}
+				</script>';
+			}
+		}
+
+		public function get_faq_answer( $block, $answer ) {
+			$count_inner_blocks = count( $block[ 'innerBlocks' ] );
+			if ( $count_inner_blocks == 0 ) {
+				// check if it contains a text
+				if ( trim( wp_strip_all_tags( $block[ 'innerHTML' ], true ) ) != '' ) {
+					// return the text without tags
+					return $answer . trim( wp_strip_all_tags( $block[ 'innerHTML'], true ) );
+				}
+				return $answer;
+			}
+
+			for ( $i = 0; $i < $count_inner_blocks; $i++ ) {
+				$partial_answer = $this->get_faq_answer( $block[ 'innerBlocks' ][ $i ], $answer );
+				$answer = $partial_answer;
+			}
+
+			return $answer;
+		}
+
+		public function render_block_accordion_faq_schema( $block_content, $block ) {
+			$attributes = $block[ 'attrs' ];
+
+			if ( isset( $attributes[ 'enableFAQ' ] ) && $attributes[ 'enableFAQ' ] ) {
+				// innerBlocks[0] is for the title
+				// retrieve stackable/column -> stackable/icon-label -> stackable/heading
+				$question = trim( strip_tags( $block[ 'innerBlocks' ][0][ 'innerBlocks' ][0][ 'innerBlocks' ][0][ 'innerHTML' ] ) );
+
+				// innerBlocks[1] is for the content
+				// content may have multiple blocks so we need to retrieve all texts from each block
+				$answer = '';
+				$answer = $this->get_faq_answer( $block[ 'innerBlocks' ][1], $answer );
+
+				$this->faq_entities[] = '{
+						"@type": "Question",
+						"name": "' . $question . '",
+						"acceptedAnswer": {
+							"@type": "Answer",
+							"text": \'' . $answer .'\'
```

Comment:
Since you are adding the `$question` and `$answer` here from inputted text, you may encounter invalid JSON formatted strings here if the question or answer contains a quotation mark. You will need to escape both variables using the `json_encode( $str )` function (I this function adds a a quotation mark at the start and end of the output, so you may need to account for that)

e.g.
```diff
- "name": "' . $question . '",
+ "name": ' . json_encode( $question ) . ',
```